### PR TITLE
[tests-only][full-ci] Improve Then step in apiWebdavEtagPropagation1 suite

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of the `When` steps.

**Suites covered:**
- `apiWebdavEtagPropagation1`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by grgprarup — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/apiWebdavEtagPropagation1/deleteFileFolder.feature [hunk 1]
```diff
@@ -16,7 +16,9 @@ Feature: propagation of etags when deleting a file or folder
     And user "Alice" has stored etag of element "/upload"
     And user "Alice" has stored etag of element "/upload/sub"
     When user "Alice" deletes file "/upload/sub/file.txt" using the WebDAV API
-    Then these etags should have changed:
+    Then the HTTP status code should be "204"
+    And as "Alice" file "/upload/sub/file.txt" should not exist
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavEtagPropagation1/deleteFileFolder.feature:20]: I don't think these changes are needed.
Here were are testing that the etags are changed, the fact that the code is correct after deleting and the file/folder is gone should be tested in other tests.

Same applies to most of the other tests below

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by individual-it — APPROVED** [review]

**Pull request merged**